api/notifyer.go

```diff
@@ -0,0 +1,61 @@
+package api
```

Comment:
This design has several problems:

* The `fsnotify` code doesn't belong in the API. The API is designed to be storage agnostic, as it can work with both file and S3 storages. The `fsnotify` feature only applies to `store.FileStore`  and it should be the responsibility of the each instance of `store.FileStore` to watch its directory tree should be in store.FileStore.
* This implementation wrongly assumes that the original images are stored using store.FileStore, when in production environments it's more likely that they are stored on AWS S3 using store.S3Store.
* It creates a watcher for every file. If the file storage has hundred of thousands of files, it would create hundred of thousands of watchers. You only need to create a watcher for each subdirectory.

Reply:
Have added required fixes